go/cmd/dolt/commands/sql.go

```diff
@@ -555,8 +555,11 @@ func getPushOnWriteHook(ctx context.Context, dEnv *env.DoltEnv) (*doltdb.PushOnW
 		return nil, err
 	}
 
-	pushHook := doltdb.NewPushOnWriteHook(ddb, dEnv.TempTableFilesDir())
-	return pushHook, nil
+	if _, val, ok := sql.SystemVariables.GetGlobal(dsqle.AsyncReplicationKey); ok && val == int8(1) {
```

Comment:
You shouldn't need the int8 cast for comparison with a constant.  The compiler should handle that.  Probably better to use a named constant